fedimint-server/src/net/api.rs

```diff
@@ -204,10 +218,23 @@ fn server_endpoints() -> Vec<ApiEndpoint<FedimintConsensus>> {
         },
         api_endpoint! {
             "/fetch_transaction",
+            async |fedimint: &FedimintConsensus, _dbtx, tx_hash: TransactionId| -> Option<TransactionStatus> {
+                debug!(transaction = %tx_hash, "Recieved request");
+
+                let tx_status = fedimint.transaction_status(tx_hash)
+                    .await;
+
+                debug!(transaction = %tx_hash, "Sending outcome");
+                Ok(tx_status)
+            }
+        },
+        api_endpoint! {
+            "/wait_transaction",
```

Comment:
I think we'll eventually either parametrize this one (only wait for tx acceptance or also outputs being finalized) or add a different endpoint for awaiting output outcomes.